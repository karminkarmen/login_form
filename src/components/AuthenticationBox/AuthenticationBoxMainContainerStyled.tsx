import styled from 'styled-components';
import { centered } from '../../configs/mixins';

const AuthenticationBoxMainContainerStyled = styled.div`
    ${centered}
    flex-direction: column;
    border-radius:  ${props => props.theme.radius.maximum};
    background: ${props => props.theme.colors.secondary};
    width: ${props => props.theme.width.mainElementWidth};
    height: ${props => props.theme.width.mainElementWidth};
`;

export default AuthenticationBoxMainContainerStyled;
