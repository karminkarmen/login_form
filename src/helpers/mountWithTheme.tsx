import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../configs/theme';

const mountWithTheme = (child: React.ReactElement) => {
    return mount(child, {
        wrappingComponent: ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    });
}

export default mountWithTheme;