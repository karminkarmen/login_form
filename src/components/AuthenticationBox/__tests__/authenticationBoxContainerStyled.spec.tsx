import React from 'react';
import mountWithTheme from '../../../helpers/mountWithTheme';
import AuthenticationBoxMainContainerStyled from '../AuthenticationBoxMainContainerStyled';

describe('<AuthenticationBoxMainContainerStyled />', () => {

    it('matches snapshot', () => {
        const wrapper = mountWithTheme(<AuthenticationBoxMainContainerStyled />);
        expect(wrapper.html()).toMatchSnapshot();
    });

});
