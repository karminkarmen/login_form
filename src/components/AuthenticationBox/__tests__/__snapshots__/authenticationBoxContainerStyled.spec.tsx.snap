// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthenticationBoxMainContainerStyled /> matches snapshot 1`] = `
"<styled.div>
  <div className=\\"sc-AxjAm csEKhz\\" />
</styled.div>"
`;
